os/fs/smartfs/smartfs_smart.c

```diff
@@ -2020,6 +1962,7 @@ int smartfs_rename(struct inode *mountpt, const char *oldrelpath, const char *ne
 		if (oldentry.dsector == newentry.dsector) {
 			/* We will not use any new entry found, we will overwrite the existing entry but with a new name */
 			strncpy(oldentry.name, newentry.name, fs->fs_llformat.namesize);
+			oldentry.prev_parent = oldentry.dsector;
```

Comment:
For newentry, finddirentry returns start sector of parent folder's sector number because there's no entry there.
But oldentry, entry is already there, it will return correct dsector.
So to compare parent directory properly, below is correct.
```
 if (oldentry.dfirst == newentry.dsector) 
```
You can test it easily put some logs..

Reply:
Mistake in pushing version heads, sorry